scripts/dts/gen_defines.py

```diff
@@ -444,6 +444,14 @@ def irq_name_alias(irq, cell_name):
             alias += "_" + str2ident(cell_name)
         return alias
 
+    def map_arm_gic_irq_type(irq, irq_type, irq_num):
+        # Maps ARM GIC IRQ (type)+(index) combo to linear IRQ number
+        if irq_type == 0: # GIC_SPI
+            irq_num = irq_num + 32
+        elif irq_type == 1: # GIC_PPI
+            irq_num = irq_num + 16
+        return irq_num
```

Comment:
Could do early returns maybe:

```python
def map_arm_gic_irq_type(irq, irq_type, irq_num):
    # Maps ARM GIC IRQ (type)+(index) combo to linear IRQ number
    if irq_type == 0: # GIC_SPI
        return irq_num + 32
    if irq_type == 1: # GIC_PPI
        return irq_num + 16
    return irq_num
```

Reply:
Python nit: Could put two spaces before `#` after code, for consistency. It's a common style in Python.

Reply:
BTW, if you think it gets clearer, feel free to move both this function and `encode_zephyr_multi_level_irq()` (and maybe `irq_name_alias()`) out of `write_irqs()` and putting them after it (I usually put top-level code at the top).

Know it was just for consistency, but `write_irqs()` might be getting a bit unwieldy. Unrelated to this PR though...

Reply:
> BTW, if you think it gets clearer, feel free to move both this function and encode_zephyr_multi_level_irq() (and maybe irq_name_alias()) out of write_irqs() and putting them after it (I usually put top-level code at the top).

I am not really a Python guy, so I will leave it to someone else who is more acquainted with the language.

Reply:
Yeah, no worries. Would've been a bit silly to force into this PR anyway.